arelle/plugin/validate/ESEF_2022/DTS.py

```diff
@@ -110,7 +113,16 @@ def lc3wordAdjust(word: str) -> str:
                         if modelConcept not in val.primaryItems:
                             extLineItemsWithoutHypercube.append(modelConcept)
                         elif not widerNarrowerRelSet.fromModelObject(modelConcept) and not widerNarrowerRelSet.toModelObject(modelConcept):
-                            if not calcRelSet.fromModelObject(modelConcept): # exclude subtotals
+                            # Reporting manual - 1.4 Anchoring -> RTS on ESEF does not set an anchoring requirement for the Notes
+                            # to the financial statements
+                            conceptRels = parentChildRelSet.toModelObject(modelConcept)
+                            conceptLinkroles = tuple(set(rel.linkrole for rel in conceptRels))
+                            conceptLinkroleRestrictedRelSet = val.modelXbrl.relationshipSet(XbrlConst.parentChild,
+                                                                                            conceptLinkroles)
+                            if not calcRelSet.fromModelObject(modelConcept) and not isChildOfNotes(modelConcept,
+                                                                                                   conceptLinkroleRestrictedRelSet,
+                                                                                                   esefNotesConcepts,
+                                                                                                   set()): # exclude subtotals
```

Comment:
This appears to be O(n^2).  It would be nice if we could pre-compute (or memoize) `childrenOfNotes` so that we can do a simple `modelConcept.qname in childrenOfNotes` here.

If we don't think this is a priority right now because the number of unanchored extensions will be small in practice, you might consider a small comment in case that becomes a problem in the future:
> This has O(extensions*presentation) running time, which could be slow if there are many unanchored extensions.  This could be improved by precomputing childrenOfNotes.

Reply:
The comment has been added.
The tests should be fine too, I have only added a comment!